### PR TITLE
Unique initializers

### PR DESCRIPTION
**Pull request by koresar**

### CODE REVIEW

File: compose.js [hunk 1]
```diff
@@ -89,7 +89,11 @@ function mergeComposable(dstDescriptor, srcComposable) {
   combineProperty('deepConfiguration', merge);
   if (Array.isArray(srcDescriptor.initializers)) {
     if (!Array.isArray(dstDescriptor.initializers)) dstDescriptor.initializers = [];
-    dstDescriptor.initializers.push.apply(dstDescriptor.initializers, srcDescriptor.initializers.filter(isFunction));
+    srcDescriptor.initializers.forEach(i => {
```
Comment (FredyC) [inline compose.js]: Would be using `reduce` better? Something like this without side effects and in my opinion somewhat easier to reason about.

```
dstDescriptor.initializers = srcDescriptor.initializers.reduce((result, init) => {
     if (isFunction(init) && !result.includes(init)) {
         result.push(init);
     }
     return result;
}, Array.isArray(dstDescriptor.initializers) ? dstDescriptor.initializers : []);
```

Comment (koresar) [inline compose.js]: I agree `reduce` is always better. This file is designed to be as readable as possible. I find `reduce` less readable than the `forEach` in this case.

Comment (FredyC) [inline compose.js]: Yea I guess it's very subjective, I even think that the line with `if (!Array.isArray(dstDescriptor.initializers))` is very hard to spot in there and it's almost unclear at a first glance why it's there. With reduce you clearly see how it's used and handled. But I am not going to fight you on this, if others like your way better too, it's fine.

Comment (koresar) [inline compose.js]: Yeah. The thing is too subjective. :)

Hey, do you want a commit access to the `stampit-org`? You'd be able to review and merge code. :)

Comment (FredyC) [inline compose.js]: Well if it were up to me to merge I would not do it, because of my opinion above :)

Comment (koresar) [inline compose.js]: Fair enough. Let me change the code to the `reduce` implementation so that you can merge it. :)

File: compose.js [hunk 2]
```diff
@@ -88,12 +88,12 @@ function mergeComposable(dstDescriptor, srcComposable) {
   combineProperty('configuration', assign);
   combineProperty('deepConfiguration', merge);
   if (Array.isArray(srcDescriptor.initializers)) {
-    if (!Array.isArray(dstDescriptor.initializers)) dstDescriptor.initializers = [];
-    srcDescriptor.initializers.forEach(i => {
-      if (isFunction(i) && dstDescriptor.initializers.indexOf(i) < 0) {
-        dstDescriptor.initializers.push(i);
+    dstDescriptor.initializers = srcDescriptor.initializers.reduce((result, init) => {
```
Comment (FredyC) [inline compose.js]: Well I hope you have tested it little, because I did not, I just wrote it from top of my head.

File: compose.js (hunk 1 above)
Comment (koresar) [inline compose.js]: Ok. Done. See the latest commit.
Also, the invitation was sent. Feel free to accept and merge the PR (use the "squash and merge" github button pls).

**Comment by FredyC** [comment]
Ugh, there are actually some tests? Where are those defined?

File: compose.js (hunk 2 above)
Comment (koresar) [inline compose.js]: I did test it. We have the special package `check-compose` which runs unit tests agains a "compose()" implementation. I have just updated the `check-compose` up to version two, - added two more tests to check that initializers do not duplicate.

**Comment by FredyC** [comment]
Ah I see, well sorry about `includes`, I kinda expected it's available since I am using it all over my projects :)

**Comment by koresar** [comment]
Totally my bug. I should have checked it agains node 0.10 myself :)

**Comment by FredyC** [comment]
Hm, Node 0.10 is so old, if it would be up to me, I would go from 0.12 above :)

**Comment by koresar** [comment]
Node 0.10 is too wide spread yet. We should deal with with.
I've just pushed the fix. Should be good to merge.

File: compose.js [hunk 3]
```diff
@@ -88,8 +88,12 @@ function mergeComposable(dstDescriptor, srcComposable) {
   combineProperty('configuration', assign);
   combineProperty('deepConfiguration', merge);
   if (Array.isArray(srcDescriptor.initializers)) {
-    if (!Array.isArray(dstDescriptor.initializers)) dstDescriptor.initializers = [];
-    dstDescriptor.initializers.push.apply(dstDescriptor.initializers, srcDescriptor.initializers.filter(isFunction));
+    dstDescriptor.initializers = srcDescriptor.initializers.reduce((result, init) => {
+      if (isFunction(init) && result.indexOf(init) < 0) {
```
Comment (unstoppablecarl) [inline compose.js]: Why check < 0 instead of === -1? Just curious

Comment (FredyC) [inline compose.js]: Using `-1` always felt weird to be as some arbitrary number. You can even reason better about it like this: "index is below 0" instead of "index is equal this weird number".

Comment (koresar) [inline compose.js]: @unstoppablecarl shorter :)

Comment (unstoppablecarl) [inline compose.js]: Makes sense. My instinct is to use `===` to be strict.
